src/EditorFeatures/Text/Extensions.SnapshotSourceText.cs

```diff
@@ -267,9 +268,9 @@ public override SourceText WithChanges(IEnumerable<TextChange> changes)
             private class ChangedSourceText : SnapshotSourceText
             {
                 private readonly SnapshotSourceText _baseText;
-                private readonly ITextSnapshot _baseSnapshot;
+                private readonly ITextImage _baseSnapshot;
```

Comment:
editor buffer only allow one way forking. they don't allow one base snapshot to fork to multiple ones like Roslyn. so only way to do that is creating new buffer. that is why it is cloning. 

_baseText.TextImage and _baseSnapshot has same content but belong to different buffer. we need _baseSnapshot and currentSnapshot to use more efficient editor's GetChanges API.